tests/localosmosis/Dockerfile

```diff
@@ -0,0 +1,52 @@
+# syntax=docker/dockerfile:1
+
+ARG BASE_IMG_TAG=nonroot
+
+# --------------------------------------------------------
+# Build
+# --------------------------------------------------------
+
+FROM golang:1.18.2-alpine3.15 as build
+
+RUN set -eux; apk add --no-cache ca-certificates build-base;
+RUN apk add git
+# Needed by github.com/zondax/hid
+RUN apk add linux-headers
+
+WORKDIR /osmosis
+COPY . /osmosis
+
+
+# CosmWasm: see https://github.com/CosmWasm/wasmvm/releases
+ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.0.0/libwasmvm_muslc.aarch64.a /lib/libwasmvm_muslc.aarch64.a
+ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.0.0/libwasmvm_muslc.x86_64.a /lib/libwasmvm_muslc.x86_64.a
+RUN sha256sum /lib/libwasmvm_muslc.aarch64.a | grep 7d2239e9f25e96d0d4daba982ce92367aacf0cbd95d2facb8442268f2b1cc1fc
+RUN sha256sum /lib/libwasmvm_muslc.x86_64.a | grep f6282df732a13dec836cda1f399dd874b1e3163504dbd9607c6af915b2740479
+
+# CosmWasm: copy the right library according to architecture. The final location will be found by the linker flag `-lwasmvm_muslc`
+RUN cp /lib/libwasmvm_muslc.$(uname -m).a /lib/libwasmvm_muslc.a
+
+RUN BUILD_TAGS=muslc LINK_STATICALLY=true make build
+
+# --------------------------------------------------------
+# Runner
+# --------------------------------------------------------
+
+FROM alpine
+
+COPY --from=build /osmosis/build/osmosisd /bin/osmosisd
+COPY /tests/localosmosis/setup.sh /setup.sh
+
+ENV HOME /osmosis
+WORKDIR $HOME
+RUN apk update \
+ && apk add jq \
+ && rm -rf /var/cache/apk/*
+RUN chmod +x /setup.sh
+RUN /setup.sh
```

Comment:
One little improvement could be not to run this script during the `docker build` phase but running as an entrypoint script of the docker container. 

In this way we can avoid rebuilding the image if we want to modify a parameter of the genesis. 
Building takes 1-2 mins while running the script should be in the order of seconds

The way to do that would be to modifying the Dockerfile like this:

```bash

...
WORKDIR $HOME
RUN apk update \
 && apk add jq \
 && rm -rf /var/cache/apk/*
RUN chmod +x /setup.sh

# Remove the RUN /setup

EXPOSE 26656
EXPOSE 26657
EXPOSE 1317

ENTRYPOINT ["/setup.sh"]
CMD ["osmosisd", "start"]
```

And in the `setup.sh`:

```bash
#!/usr/bin/env sh
set -eu

<STANDARD CONTENT>

exec "$@" 
```

The `exec` part will execute the `CMD` part of the Dockerfile after is has executed the script

Reply:
The only problem with this is the user would be recreating the genesis process on every start right? This process should only be done once at compilation time I think, and if the user wanted to modify a genesis param, they are probably better off recompiling anyway right?